presto-hive/src/main/java/com/facebook/presto/hive/metastore/HiveColumnStatistics.java

```diff
@@ -49,43 +58,172 @@ public HiveColumnStatistics(
         this.distinctValuesCount = distinctValuesCount;
     }
 
+    @JsonProperty
+    @JsonTypeInfo(use = CLASS, property = "@lowValueClass")
     public Optional<?> getLowValue()
     {
         return lowValue;
     }
 
+    @JsonProperty
+    @JsonTypeInfo(use = CLASS, property = "@highValueClass")
     public Optional<?> getHighValue()
     {
         return highValue;
     }
 
+    @JsonProperty
     public OptionalLong getMaxColumnLength()
     {
         return maxColumnLength;
     }
 
+    @JsonProperty
     public OptionalDouble getAverageColumnLength()
     {
         return averageColumnLength;
     }
 
+    @JsonProperty
     public OptionalLong getTrueCount()
     {
         return trueCount;
     }
 
+    @JsonProperty
     public OptionalLong getFalseCount()
     {
         return falseCount;
     }
 
+    @JsonProperty
     public OptionalLong getNullsCount()
     {
         return nullsCount;
     }
 
+    @JsonProperty
     public OptionalLong getDistinctValuesCount()
     {
         return distinctValuesCount;
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        HiveColumnStatistics that = (HiveColumnStatistics) o;
+        return Objects.equals(lowValue, that.lowValue) &&
+                Objects.equals(highValue, that.highValue) &&
+                Objects.equals(maxColumnLength, that.maxColumnLength) &&
+                Objects.equals(averageColumnLength, that.averageColumnLength) &&
+                Objects.equals(trueCount, that.trueCount) &&
+                Objects.equals(falseCount, that.falseCount) &&
+                Objects.equals(nullsCount, that.nullsCount) &&
+                Objects.equals(distinctValuesCount, that.distinctValuesCount);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(lowValue, highValue, maxColumnLength, averageColumnLength, trueCount, falseCount, nullsCount, distinctValuesCount);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("lowValue", lowValue)
+                .add("highValue", highValue)
+                .add("maxColumnLength", maxColumnLength)
```

Comment:
I think this `toString` method is not going to be used for other than debugging purposes. I find it useful when i see in a debugger what is actually there as a result of  `toString`. Printing a field as null is misleading. Because it may be null than in two different cases. When the optional is empty, and when the field itself is null (what surprisingly easy to achieve for this very class, since there are no null checks, but i'm gonna fix that, that's not the point).